test/parallel/test-permission-net-websocket.js

```diff
@@ -0,0 +1,18 @@
+// Flags: --permission --allow-fs-read=*
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto) {
+  common.skip('no crypto');
+}
+
+const wsUrl = 'ws://nodejs.org';
+
+const ws = new WebSocket(wsUrl);
+
+ws.addEventListener('open', common.mustNotCall('WebSocket connection should be blocked'));
+// WebSocket implementation doesn't expose the Node.js specific errors
+// so ERR_ACCESS_DENIED won't be "caught" explicitly.
+// For now, let's just assert the error
+// TODO(rafaelgss): make this test more comprehensive
+ws.addEventListener('error', common.mustCall());
```

Comment:
This was fixed here: https://github.com/nodejs/undici/pull/4274 so once that lands we can update this test to assert `cause`